src/app/common/hooks/use-hover-with-children.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useState } from 'react';
+
+interface HoverBind {
+  onMouseEnter(event: React.MouseEvent<HTMLElement, MouseEvent>): void;
+  onMouseLeave(event: React.MouseEvent<HTMLElement, MouseEvent>): void;
+}
+
+export const useHoverWithChildren = (): [boolean, HoverBind] => {
+  const [isHovered, setIsHovered] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isHovered, setIsHovered] = useState(false);
```

nit: `boolean` is inferred. Saving myself from having to do it in the lint branch.